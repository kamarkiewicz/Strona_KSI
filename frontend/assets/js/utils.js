
// Access image field from backend
function getImage (image) {
  if ((image || {}).path) {
    // path may be relative URL
    let isAbsolute = /^(?:[a-z]+:)?\/\//i.test(image.path)
    image.path = (isAbsolute ? '' : process.env.API_URL) + image.path
  }
  return image
}

export { getImage };
