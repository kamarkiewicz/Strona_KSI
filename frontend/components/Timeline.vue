<template>
  <div class="timeline container">

    <article class="row dotline" v-for="el in entries" :key="el.id">
      <div class="p-4 col-md-12">
        <div class="row dotline-row pl-2">
          <header class="col-md-6 col-lg-3">
            <time class="text-secondary font-weight-bold"
                  :datetime="new Date(el.date).toISOString().substring(0, 10)"
                  v-text="$d(el.date)"
            ></time>
            <h2 class="pt-2 text-primary font-weight-bold text-uppercase" v-text="el.title"></h2>
          </header>
          <section class="col-md-6 py-3 col-lg-4">
            <img class="img-fluid d-block mx-auto" :src="el.image.src" :alt="el.image.title" width="290">
          </section>
          <section class="col-md-12 col-lg-5 py-1" v-html="el.content"></section>
        </div>
      </div>
    </article>

  </div>
</template>

<script>
export default {
  props: {
    entries: {
      type: Array,
      required: true
    },
  },
}
</script>

<style lang="scss">

$dotline-color: #bbb;

.timeline {
  h2 {
    font-size: 1.1em;
    line-height: 1.3;
  }
}

.dotline {
  /// line
  border-color: $dotline-color;
  border-left: solid 1px;
  &:last-child {
    border-left: none;
  }
  border-left-color: $dotline-color;
  position: relative;

  /// dots
  &::before {
    background-color: $dotline-color;
    position: absolute;
    content: '';
    border-radius: 100%;
    width: 1em;
    height: 1em;
    left: -0.5em;
    top: 0;
  }

  /// content row
  .dotline-row {
    margin-top: -1.75em;
    margin-bottom: 1.75em
  }
}

</style>
