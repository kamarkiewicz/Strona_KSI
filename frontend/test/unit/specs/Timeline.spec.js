import { mount } from '@vue/test-utils'
import Timeline from '~/components/Timeline.vue'
import { Image } from '~/assets/js/utils'

describe('Timeline', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Timeline, {
      mocks: {
        $t: (label) => { return label },
        $d: (label) => { return label },
      },
      propsData: {
        entries: [
          {
            id: 1,
            date: "09.04.16",
            title: "Magna etiam sed lorem ipsum dolor amet",
            image: "https://picsum.photos/g/300/180?image=81",
            content: "Phasellus in ante vitae nibh porttitor tempus vitae ut ante. Vestibulum blandit dolor elementum viverra. Sed quat a diam, aliquet tempus felis. Phasellus et magna vitae nibh porttitor tempus vitae.",
          },
          {
            id: 2,
            date: "08.28.16",
            title: "Sed feugiat et mauris adipiscing dolor",
            image: "https://picsum.photos/g/300/180?image=82",
            content: "Phasellus in ante vitae nibh porttitor tempus vitae ut ante. Vestibulum blandit dolor elementum viverra. Sed quat a diam, aliquet tempus felis. Phasellus et magna vitae nibh porttitor tempus vitae.",
          },
          {
            id: 3,
            date: "08.22.16",
            title: "Tellus magna dolore sed elementum",
            image: "https://picsum.photos/g/300/180?image=83",
            content: "Phasellus in ante vitae nibh porttitor tempus vitae ut ante. Vestibulum blandit dolor elementum viverra. Sed quat a diam, aliquet tempus felis. Phasellus et magna vitae nibh porttitor tempus vitae.",
          },
        ],
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
