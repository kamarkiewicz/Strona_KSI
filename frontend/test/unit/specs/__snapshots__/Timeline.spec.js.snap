// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
<section class="timeline">
  <section>
    <header>
      <p>09.04.16</p>
      <h4>Magna etiam sed lorem ipsum dolor amet</h4>
    </header>
    <div class="image">
      <img src="https://picsum.photos/g/264/134?image=81" alt="">
    </div>
    <div class="content">
      <p>Phasellus in ante vitae nibh porttitor tempus vitae ut ante. Vestibulum blandit dolor elementum viverra. Sed quat a diam, aliquet tempus felis. Phasellus et magna vitae nibh porttitor tempus vitae.</p>
      <ul class="actions">
        <li>
          <a href="#" class="button">More</a>
        </li>
      </ul>
    </div>
  </section>
  <section>
    <header>
      <p>08.28.16</p>
      <h4>Sed feugiat et mauris adipiscing dolor</h4>
    </header>
    <div class="image">
      <img src="https://picsum.photos/g/264/134?image=82" alt="">
    </div>
    <div class="content">
      <p>Phasellus in ante vitae nibh porttitor tempus vitae ut ante. Vestibulum blandit dolor elementum viverra. Sed quat a diam, aliquet tempus felis. Phasellus et magna vitae nibh porttitor tempus vitae.</p>
      <ul class="actions">
        <li>
          <a href="#" class="button">More</a>
        </li>
      </ul>
    </div>
  </section>
  <section>
    <header>
      <p>08.22.16</p>
      <h4>Tellus magna dolore sed elementum</h4>
    </header>
    <div class="image">
      <img src="https://picsum.photos/g/264/134?image=83" alt="">
    </div>
    <div class="content">
      <p>Phasellus in ante vitae nibh porttitor tempus vitae ut ante. Vestibulum blandit dolor elementum viverra. Sed quat a diam, aliquet tempus felis. Phasellus et magna vitae nibh porttitor tempus vitae.</p>
      <ul class="actions">
        <li>
          <a href="#" class="button">More</a>
        </li>
      </ul>
    </div>
  </section>
</section>
`;
